### PR TITLE
Add unit test for DA task

### PR DESCRIPTION
**Pull request by shenkeyao**
Add a unit test for the DA task, to test the handling of proposal, vote and view change events.

Closes https://github.com/EspressoSystems/HotShot/issues/1493.

### CODE REVIEW

File: task-impls/src/da.rs [hunk 1]
```diff
@@ -261,6 +261,9 @@ where
 
                 // Allow a DA proposal that is one view older, in case we have voted on a quorum
                 // proposal and updated the view.
+                // `self.cur_view` should be at least 1 since there is a view change before getting
```
Comment (elliedavidson) [inline task-impls/src/da.rs:264]: Good comment!

File: testing/tests/consensus_task.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@
 // use hotshot::traits::election::static_committee::StaticElectionConfig;
 // use hotshot::traits::election::static_committee::StaticVoteToken;
 // use hotshot::traits::election::vrf::JfPubKey;
-// use hotshot::traits::implementations::MemoryCommChannel;
+// use hotshot::traits::implementations::MemoryCommChannefl;
```
Comment (elliedavidson) [inline testing/tests/consensus_task.rs:14]: Is this a typo? :)

**Review by elliedavidson — CHANGES_REQUESTED** [review]
Just the one typo

File: testing/tests/consensus_task.rs (hunk 2 above)
Comment (shenkeyao) [reply testing/tests/consensus_task.rs]: Yes, fixed!

**Review by elliedavidson — DISMISSED** [review]

**Review by elliedavidson — APPROVED** [review]

**Pull request merged**